pkg/pgmodel/cache/series_cache.go

```diff
@@ -2,70 +2,81 @@
 // Please see the included NOTICE for copyright information and
 // LICENSE for a copy of the license.
 
-package model
+package cache
 
 import (
 	"encoding/binary"
 	"fmt"
 	"math"
 	"sort"
 	"strings"
-	"sync"
 
 	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/timescale/promscale/pkg/clockcache"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
 	"github.com/timescale/promscale/pkg/prompb"
 )
 
-// Labels stores a labels.Labels in its canonical string representation
-type Labels struct {
-	Names      []string
-	Values     []string
-	MetricName string
-	str        string
+const DefaultSeriesCacheSize = 500000
+
+type SeriesCache interface {
+	Reset()
+	GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error)
+	Len() int
+	Cap() int
+}
```

Comment:
This interface is used only once (which does not justify its need).
Can we not have a generic interface, something like 

```
type Cache interface {
  Len() int
  Cap() int
  Reset()
}
```
And this should be inside `pkg/clockcache` and then imported from there to every other place. This would be awesome for the cache implementation. Else, I see 3 different kinds of interfaces - `SeriesCache`, `MetricCache`, `LabelsCache` and all have a singlar usage (except MetricCache, which can be inside the Cache interface as I said), which is not great.

Reply:
We have already seperate interfaces for MetricCache and and LabelCache, so I'm just following an established convention. If we want to change, let's discuss but not for this PR

Reply:
We should change that, sure for the other PR, but then we should add a note in the project board.

Reply:
The MetricCache interface has methods that simplify calling the underlying clock cache,
so we don't have to convert string to []interface{}
```
type MetricCache interface {
	Get(metric string) (string, error)
	Set(metric string, tableName string) error
	Len() int
	Cap() int
}

type LabelsCache interface {
	GetValues(keys []interface{}, valuesOut []interface{}) (numFound int)
	InsertBatch(keys []interface{}, values []interface{}) (numInserted int)
	Len() int
	Cap() int
}
```
And the SeriesCache will have a `GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error)` that does not apply to the other caches, we can put the Len, Cap and Reset methods to an interface in the clockcache package, but we either keep these 3 (while extending the clockcache.Cache interface) , or define a new SuperCacheInterface that has methods for all of them, or write new wrappers around the single interface that will hold the logic for the now distinct methods

Reply:
We can end up with 
```go
type SeriesCache interface {
	clockcache.CacheInterface
	GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error)
}

type seriesCacheImpl struct {
	*clockcache.CacheStruct
}

func GetSeriesFromProtos(labelPairs []prompb.Label) (series *model.Series, metricName string, err error){
  panic("implement me")
}
```